tests/environment.bats

```diff
@@ -2,9 +2,63 @@
 
 load '/usr/local/lib/bats/load.bash'
 
-@test "Exposes 'hash_file' command" {
+@test "Exposes 'add_host_to_ssh_known_hosts' command" {
   source "$PWD/hooks/environment"
-  which hash_file > /dev/null
+  which add_host_to_ssh_known_hosts > /dev/null
+  return $?
+}
+
+@test "Exposes 'add_ssh_key_to_agent' command" {
+  source "$PWD/hooks/environment"
+  which add_ssh_key_to_agent > /dev/null
+  return $?
+}
```

Comment:
I am all for more unit tests, but I can also see us forgetting to add entries to those tests when adding new commands in `bin/` in the future (I for one know I have, and that's also probably the reason why you had to add so many missing ones here).

What do you think about re-writing all those tests about "Exposes x command" into a single test which loops on all commands instead, so that it ends up being a dynamic test with no need for update on new commands?

Something like:
```sh
@test "Exposes all commands in bin/" {
  source "$PWD/hooks/environment"
  for cmd in "$PWD/bin"/*; do
    which "$(basename "$cmd")" > /dev/null
  done
}
```
From what I understand [from the `bats` README](https://github.com/bats-core/bats-core/tree/master#bats-core-bash-automated-testing-system), this should be enough (without the need for `return $?`), as the test will fail as soon as any command has a non-zero exit code:

> Test cases consist of standard shell commands. Bats makes use of Bash's errexit (set -e) option when running test cases. If every command in the test case exits with a 0 status code (success), the test passes. In this way, each line is an assertion of truth.

Reply:
Thanks! This is a great sugggestion! And tbh, I wasn't aware of the tests either. It was only when I started digging deeper into the repo/plugin that I noticed there were some missing tests for most of the commands.

But I agree with you, I think a single test that iterates over all the commands would be:

- cleaner
- one less thing to remember
- suffient as the test will fail if any of the commands return a non-zero exit code

Reply:
Suggestion applied in 85717fb8c9027de353ea642b2a6171c13f657fcd